cfpb/cfgov-refresh
# Replace disabled links with headings in mega menu

Fixes https://GHE/CFGOV/platform/issues/2593 and https://GHE/CFGOV/platform/issues/2607

Headings are brought back into the mega menu to add: 
- navigable landmarks to the page for screen reader users
- additional context for sighted mobile users when navigating through the submenus.


## Additions

- a new `<h2>` heading element is added to the mobile menu and is hidden from sighted users, but read aloud by screen readers. This heading is used as an aria label and landmark to describe the purpose of the `<nav>` element. it replaces the previous way we described the `<nav>` element which was another aria attribute. 
- benefits of this change: 
  - screen reader users still hear a landmark/label for nav element, and also now can get to the navigation by navigating through the page headings. 
  - gives a parent heading element to the new `<h3>` headings that I added for overview text, and to the existing `<h4>` elements (see details below)

## Removals

-

## Changes

- Overview links are only link elements if they have a real value, otherwise they default to heading elements (`<h3>`)
- `<h3>` and `<h4>` elements are now nested inside an `<h2>` (see "Additions" above)
- brought back the `<h4>` element for the "Money Topics", "Guides", etc text. Since these bits of text describe the content after them, it makes sense to keep them as headings. See https://GHE/CFGOV/platform/issues/2593#issuecomment-165887 for more detail and research findings on this approach.
- CSS was updated to accommodate the above markup changes while keeping the design the same.
- The only visible design difference: on the mobile mega menu, the overview text for the Consumer Tools and Practitioner Resources menus is now visible as regular text (instead of a disabled link).

## Testing

1. check out branch, run `gulp styles`
1. `./runserver.sh` and visit http://localhost:8000/
1. optional:  navigate through the mega menu at small and large screen sizes with a mouse
1. optional:  navigate using keyboard: tab to move through links

### Testing with desktop voiceover
Most importantly: test using voiceover on desktop. follow instructions in the Web Accessibility Audit, Section B, "How to test" or do the following short version:
  1. cmd+F5 to trigger voiceover
  1. with focus on the webpage (tab into the web content in Chrome or click on the page), use keyboard shortcut *control+option+U* to trigger the Rotor interface
  1. use the right arrow key to get to the "Headings" list in the Rotor
  1.  use the down arrow key to select "2. Main menu" (it should be first in the list of Headings) and hit enter. this will bring you to that element on the page
  1. *control+option+A* to read the mega menu elements in order
  1. use *enter* key to select a submenu like "Consumer Tools" and *control+option+A* again to keep reading the submenu items
  1. confirm the headings within each submenu are read aloud when doing this
 


## Screenshots
![screen shot 2018-06-05 at 5 00 07 pm](https://user-images.githubusercontent.com/702526/41005293-dfa8c6c8-68ea-11e8-902b-c1cf19f47227.png)
![screen shot 2018-06-05 at 4 59 59 pm](https://user-images.githubusercontent.com/702526/41005295-dfc06526-68ea-11e8-958f-25f3426ebb43.png)
![screen shot 2018-06-05 at 4 57 29 pm](https://user-images.githubusercontent.com/702526/41005296-dfd5aca6-68ea-11e8-8513-35742f3422b0.png)
![screen shot 2018-06-05 at 4 57 23 pm](https://user-images.githubusercontent.com/702526/41005297-dfee6ac0-68ea-11e8-9835-cb2f0f0fe018.png)

Here are some screen recordings of navigating the desktop menu with the headings using VoiceOver.

- [Navigating the desktop mega menu, "Practitioner Resources" section](https://drive.google.com/file/d/18a7FDkVK4KZdw7Aae6wrgDNHj24KfIfI/view?usp=sharing)
- [Navigating the desktop mega menu, "About Us" section](https://drive.google.com/file/d/1Sa9nC5aCjOqHy5DRBszK6PH5eCujlIIu/view?usp=sharing)

## Notes

-

## Todos

- [x] test in diff browsers using Sauce Labs
- [x] add quicktime recording of how this works on Desktop VoiceOver
- [ ] get help testing in iOs VoiceOver - either it's totally broken or I don't know enough about how to use mobile VO (prob a mix of both)

## Checklist


## Testing checklist

### Browsers


### Accessibility


### Other
